### DISCUSSION TITLE
Question about Zed extension development - Combining and processing file content

### DISCUSSION THREAD

**Discussion by KennyxLan** [zed-industries/zed#16872] [Feature Requests]
Hi everyone,

I'm fairly new to Zed and extension development. I'm interested in writing an extension that can combine the contents of multiple files, process them, and then display the results in a dedicated view.

To achieve this, I imagine I'll need APIs for:

- Reading from and writing to files
- Creating user interfaces within the Zed editor buffer

I've looked through the Zed Extension API documentation ( [zed_extension_api](https://docs.rs/zed_extension_api/latest/zed_extension_api/index.html) ), but haven't found anything directly related to these functionalities.

Does Zed currently offer APIs for these tasks? If so, could you point me in the right direction for the documentation?

Thanks in advance for your help!

**Comment by bajrangCoder** [comment]
At the moment, developing these types of extensions isn't supported. Currently, you can only create the following extensions: [Extension Capabilities](https://zed.dev/docs/extensions/developing-extensions#extension-capabilities).

**Reply by KennyxLan** [reply]
Thanks for the information. I understand that developing extensions with the capabilities I need isn't currently supported. I'll explore the available Zed features to see what I can achieve.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
